agent-control/src/cli/install.rs

```diff
@@ -121,10 +122,16 @@ pub fn apply_resources(
     let k8s_client = try_new_k8s_client()?;
     let maybe_helm_release = k8s_client
         .get_dynamic_object(&helmrelease_v2_type_meta(), release_name, namespace)
-        .map_err(|err| {
-            CliError::ApplyResource(format!(
+        .map_err(|err| match err {
+            // Specifying an invalid version of `agent-control-cd` throws an error showing that something is wrong with the api.
+            // It can be hard to understand without prior knowledge, thus we simplify the error message for that specific variant.
+            K8sError::MissingAPIResource(_) => CliError::ApplyResource(format!(
+                "could not get helmRelease with name {release_name} and version {}",
```

Comment:
super-nit: just to match the name of the resource

```suggestion
                "could not get HelmRelease with name {release_name} and version {}",
```